examples/libsodium/CMakeLists.txt

```diff
@@ -0,0 +1,13 @@
+cmake_minimum_required(VERSION 3.0)
+project(libsodium-example)
+
+set(HUNTER_CONFIGURATION_TYPES Release) #Release or Debug
```

Comment:
> the library from one build type will be overwritten by the library for the other type when installing

in cmake libraries should have different names but headers and common config files should be identical. note that you not only have to rename library but patch your `*Config.cmake.in` template accordingly.

see https://github.com/forexample/package-example#install-foo

in general it's quite tricky to do for non-cmake files, probably setting default build type to Release is easiest applieable workaround, see https://github.com/ruslo/hunter/wiki/usr.adding.new.package#build-types (note that it's not the same as using HUNTER_CONFIGURATION_TYPES, so we will not have issues I've described)